### PR TITLE
4.0 packages

### PR DESCRIPTION
**Pull request by BrainforgeUK**
Pull Request for Issue #17585  .

### Summary of Changes

Added -- Select Package -- list to filter in extension manager, module manager and plugin manager.

### Testing Instructions

Goto .../administrator/index.php?option=com_installer&view=manage
Click on filter options

### Expected result

Select a package and only extensions within the package (plus the package itself) will be displayed.
Similar result with module and plugin managers.

### CODE REVIEW

File: administrator/components/com_installer/forms/filter_manage.xml [hunk 1]
```diff
@@ -45,6 +45,15 @@
 			<option value="">COM_INSTALLER_VALUE_FOLDER_SELECT</option>
 		</field>
 
+		<field
+				name="package"
+				type="package"
+				label="COM_INSTALLER_VALUE_PACKAGE_SELECT"
+				onchange="this.form.submit();"
+		>
```
Comment (Quy) [inline administrator/components/com_installer/forms/filter_manage.xml]:
```suggestion
			>
```

File: administrator/components/com_installer/forms/filter_manage.xml [hunk 2]
```diff
@@ -45,6 +45,15 @@
 			<option value="">COM_INSTALLER_VALUE_FOLDER_SELECT</option>
 		</field>
 
+		<field
+				name="package"
```
Comment (Quy) [inline administrator/components/com_installer/forms/filter_manage.xml]: Reduce indentation for lines 49-52.

File: administrator/components/com_installer/Field/PackageField.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Form\Field\ListField;
+use Joomla\Component\Installer\Administrator\Helper\InstallerHelper;
+
+/**
+ * Package field.
+ *
+ * @since  3.5
```
Comment (Quy) [inline administrator/components/com_installer/Field/PackageField.php]: Change all instances of `@since` to `__DEPLOY_VERSION__`.

**Comment by brianteeman** [comment]
I can see the use in the extension manager but what is the use case for modules and plugins especially as we don't display the package information there

File: administrator/components/com_installer/forms/filter_manage.xml (hunk 2 above)
Comment (BrainforgeUK) [reply administrator/components/com_installer/forms/filter_manage.xml]: Changed.

File: administrator/components/com_installer/Field/PackageField.php (hunk 3 above)
Comment (BrainforgeUK) [reply administrator/components/com_installer/Field/PackageField.php]: Changed PackageField.php, ModulesPackageField.php and PluginPackageField.php
Also changed the new package function in the helper for each of these components.

**Comment by BrainforgeUK** [comment]
> 
> 
> I can see the use in the extension manager but what is the use case for modules and plugins especially as we don't display the package information there

The original issue mentioned having links to plugin edit in manager listing. Not implemented for 4.0 (but might still be a good idea). Being able to filter plugins and modules by package useful when all one is interested in are package related extensions - no other clutter!

**Comment by richard67** [comment]
@BrainforgeUK Can it be that file "administrator/language/en-GB/en-GB.com_installer.ini" in your PR is based on an outdated version? When checking the differences in the modified files of your PR, I see for this file more changes than those related to what you PR does, and some of the changes make things wrong, e.g. "a update" instead of "an update", or "upload extensions into the installer" instead of "upload extensions with the installer.".

**Comment by brianteeman** [comment]
>  Being able to filter plugins and modules by package useful when all one is interested in are package related extensions - no other clutter!

As there is no way for me to know if a plugin or modules belongs to a package  on that screen I dont see how I would even know what to filter by.

This really sounds like a way of solving a problem with an extension by modifying the core to me.

**Comment by Schmidie64** [comment]
I have tested this item :red_circle: unsuccessfully on 2d51883abaad3001bb29b3d003ce163caa173eab<br /><br />after applying the patch, i got a broken site. 
image

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25105">issues.joomla.org/tracker/joomla-cms/25105</a>.</sub>

**Comment by richard67** [comment]
@Schmidie64 Sure, because PR is not really ready for test. Due to changes in the 4.0-dev branch which have happened meanwhile, this patch needs to be updated, and conflicts need to be resolved. When a patch shows it has conflicts at the bottom of the GitHub page, it makes no sense to test it because it will very likely fail. So I recommend you change back your tes result to "I have not tested this item" in the issue tracker.

@BrainforgeUK Could you update your PR to latest 4.0-dev and solve the conflicts?

**Comment by BrainforgeUK** [comment]
Made changes and commited and pushed to github.
https://github.com/BrainforgeUK/joomla-cms/tree/4.0-packages

Cannot see them in the pull-request.

**Comment by roland-d** [comment]
@BrainforgeUK I think your last push did come through just not so fast as I can see the commits above your message. Do you want to fix the conflicts and pick this up or should the pull request be deleted?

**Pull request closed without merging**